spec/index.bs

```diff
@@ -7364,14 +7380,40 @@ rendering operations have been flushed to the drawing buffer.
         that created this swap chain next.
 
         <div algorithm="GPUSwapChain.getCurrentTexture">
-            **Called on:** {{GPUSwapChain}} this.
+            **Called on:** {{GPUSwapChain}} |this|.
 
             **Returns:** {{GPUTexture}}
 
-            Issue: Describe {{GPUSwapChain/getCurrentTexture()}} algorithm steps.
+            1. Set |this|.{{GPUSwapChain/[[dirty]]}} to `true`.
+            1. Return |this|.{{GPUSwapChain/[[currentTexture]]}}.
         </div>
 </dl>
 
+<div algorithm>
+    During the "update the rendering [of the] `Document`" step of the "[=Update the rendering=]"
+    HTML processing model, the associated {{HTMLCanvasElement}} |canvas| for each {{GPUSwapChain}}
+    |swapChain| should <dfn abstract-op>present the swap chain content to the canvas</dfn> by running
+    the following steps:
+
+        1. If |swapChain|.{{GPUSwapChain/[[dirty]]}} is `false`, stop.
+        1. Let |displayTexture| be |swapChain|.{{GPUSwapChain/[[currentTexture]]}}.
+        1. Let |swapChain|.{{GPUSwapChain/[[currentTexture]]}} be a new {{GPUTexture}} with a
+            {{GPUTexture/[[textureSize]]}} of |canvas|.width and |canvas|.height, {{GPUTexture/[[format]]}}
+            of |swapChain|.{{GPUSwapChain/[[format]]}}, and {{GPUTexture/[[textureUsage]]}} of
+            |swapChain|.{{GPUSwapChain/[[textureUsage]]}}. If a previously presented texture from
+            |swapChain| matches the required criteria it's GPU memory may be re-used.
+        1. Ensure that all rendering operations have been flushed to |displayTexture|.
+        1. Update |canvas| with the contents of |displayTexture|.
+        1. |displayTexture| must act as if {{GPUTexture/destroy()}} were called on it without
+            releasing the associated GPU memory.
+        1. Set |swapChain|.{{GPUSwapChain/[[dirty]]}} to be `false`.
+</div>
+
+Note: Developers can expect that the same {{GPUTexture}} object will be returned by every call to
+{{GPUSwapChain/getCurrentTexture()}} made within the same callback, and additionally that the
+same {{GPUTexture}} object will be returned in each `requestAnimationFrame()` callback run as part
+of the same frame.
```

Comment:
Not just inside rAFs, but within any code run within the same frame.

"... every call to getCurrentTexture() made within the same frame (i.e. between invocations of [=Update the rendering=])."